fs_storage/models/fs_storage.py

```diff
@@ -302,6 +307,28 @@ def _recursive_add_odoo_storage_path(self, options: dict) -> dict:
             self._recursive_add_odoo_storage_path(target_options)
         return options
 
+    def _eval_options_from_env(self):
+        values = {}
+        for key, value in self.json_options.items():
+            if value.startswith("$"):
```

Comment:
Value may not be a string. If this is JSON it could be a list or a dict, or a number.
Recursive dict should be handled, I think. In the help text option there is such an example.